src/_fixErroneousExpansionMdx.test.ts

```diff
@@ -0,0 +1,484 @@
+              )
+            )
+          ),
+          Crossjoin(
+            [Currency].[Currency].[ALL].[AllMember].[EUR],
+            Descendants(
+              {
+                [Geography].[City].[ALL].[AllMember]
+              },
+              [Geography].[City].[City]
+            )
+          )
+        )
+      ) ON ROWS
+      FROM [EquityDerivativesCube]
+      CELL PROPERTIES VALUE, FORMATTED_VALUE, BACK_COLOR, FORE_COLOR, FONT_FLAGS`);
+
+    const updatedMdx = _fixErroneousExpansionMdx(mdx, cube);
+
+    expect(stringify(updatedMdx, { indent: true })).toMatchInlineSnapshot(`
+      "SELECT
+        NON EMPTY Hierarchize(
+          Crossjoin(
+            Hierarchize(
+              DrilldownLevel(
+                [Currency].[Currency].[ALL].[AllMember]
+              )
+            ),
+            Hierarchize(
+              DrilldownLevel(
+                [Geography].[City].[ALL].[AllMember]
+              )
+            )
+          )
+        ) ON ROWS
+        FROM [EquityDerivativesCube]
+        CELL PROPERTIES VALUE, FORMATTED_VALUE, BACK_COLOR, FORE_COLOR, FONT_FLAGS"
+    `);
+  });
+
+  it("removes the problematic part of the Mdx of a widget where a user collapsed and then re-expanded a member in ActiveUI 4 - even if one of the involved hierarchies is slicing", () => {
+    const mdx = parse<MdxSelect>(`SELECT
+      NON EMPTY Hierarchize(
+        Union(
+          Crossjoin(
+            Hierarchize(
+              [Time].[HistoricalDates].[AsOfDate].Members
+            ),
+            Hierarchize(
+              DrilldownLevel(
+                [Currency].[Currency].[ALL].[AllMember]
+              )
+            )
+          ),
+          Crossjoin(
+            [Time].[HistoricalDates].[AsOfDate].[2021-12-10],
+            Descendants(
+              {
+                [Currency].[Currency].[ALL].[AllMember]
+              },
+              [Currency].[Currency].[Currency]
+            )
+          )
+        )
+      ) ON ROWS
+      FROM [EquityDerivativesCube]
+      CELL PROPERTIES VALUE, FORMATTED_VALUE, BACK_COLOR, FORE_COLOR, FONT_FLAGS`);
+
+    const updatedMdx = _fixErroneousExpansionMdx(mdx, cube);
+
+    expect(stringify(updatedMdx, { indent: true })).toMatchInlineSnapshot(`
+      "SELECT
+        NON EMPTY Hierarchize(
+          Crossjoin(
+            Hierarchize(
+              [Time].[HistoricalDates].[AsOfDate].Members
+            ),
+            Hierarchize(
+              DrilldownLevel(
+                [Currency].[Currency].[ALL].[AllMember]
+              )
+            )
+          )
+        ) ON ROWS
+        FROM [EquityDerivativesCube]
+        CELL PROPERTIES VALUE, FORMATTED_VALUE, BACK_COLOR, FORE_COLOR, FONT_FLAGS"
+    `);
+  });
+
+  it("removes the problematic parts of the Mdx of a widget where a user collapsed and then re-expanded 2 different members in ActiveUI 4 (collapsing all of them, then re-expanding all of them)", () => {
```

Comment:
While testing the new function, I discovered that collapsing and re-expanding actually contains several bugs in ActiveUI 4.

If you do it on several members, then:
- collapsing then re-expanding each one before moving on to the next yields a bad mdx (see test case above)
- collapsing several in a row, then re-expanding them all yields an even worst mdx (this test case)